libs/librepcb/common/signalslot.h

```diff
@@ -114,8 +121,16 @@ class Signal {
    * @param args  Arguments passed to the slots
    */
   void notify(Args... args) noexcept {
-    for (auto slot : mSlots) {
-      slot->mCallback(mSender, args...);
+    // Note: A "foreach" loop with a Qt container first creates an implicitly
+    // shared copy of the container. This is very important since the callback
+    // might modify the container while iterating over it! With std containers
+    // this would be much more complicated (or less efficient).
```

Comment:
Maybe link to https://doc.qt.io/qt-5/containers.html#foreach for interested people :slightly_smiling_face:

Reply:
Done :)